app/services/prior_authority/assessment_syncer.rb

```diff
@@ -1,82 +1,117 @@
-          adjustment_comment: cost_data['adjustment_comment']
-        )
-      end
+    def sync_allowances
+      sync_primary_quote(application.primary_quote, data['quotes'].find { _1['primary'] })
 
-      def sync_further_info_request(application, app_store_record)
-        data = app_store_record['application']
-        further_info_required = data['updates_needed'].include? 'further_information'
-        info_correct_required = data['updates_needed'].include? 'incorrect_information'
-        application.update(
-          further_information_explanation: further_info_required ? data['further_information_explanation'] : nil,
-          incorrect_information_explanation: info_correct_required ? data['incorrect_information_explanation'] : nil,
-          resubmission_deadline: data['resubmission_deadline'],
-          resubmission_requested: DateTime.current
+      application.additional_costs.each do |additional_cost|
+        sync_additional_cost(
+          additional_cost,
+          data['additional_costs'].find { _1['id'] == additional_cost.id },
         )
       end
+    end
 
-      def build_form(application, record, form_class, data)
-        form_class.new(
-          data.slice(*form_class.attribute_names).merge(application:, record:)
-        )
-      end
+    def sync_primary_quote(quote, quote_data)
+      base_cost_form = build_form(application, quote, Steps::ServiceCostForm, quote_data)
+      travel_cost_form = build_form(application, quote, Steps::TravelDetailForm, quote_data)
+
+      quote.update(
+        base_cost_allowed: base_cost_form.total_cost,
+        travel_cost_allowed: travel_cost_form.total_cost,
+        travel_adjustment_comment: quote_data['travel_adjustment_comment'],
+        service_adjustment_comment: quote_data['adjustment_comment']
+      )
+    end
+
+    def sync_additional_cost(cost, cost_data)
+      cost_form = build_form(application, cost, Steps::AdditionalCosts::DetailForm, cost_data)
+
+      cost.update(
+        total_cost_allowed: cost_form.total_cost,
+        adjustment_comment: cost_data['adjustment_comment']
+      )
+    end
+
+    def sync_sent_back_request
+      application.update(
+        incorrect_information_explanation: incorrect_information_explanation,
+        resubmission_deadline: resubmission_deadline,
+        resubmission_requested: DateTime.current
+      )
+
+      return unless further_info_required?
+
+      sync_further_info_request
+    end
+
+    def further_info_required?
+      data['updates_needed'].include?('further_information')
+    end
+
+    def incorrect_information_explanation
+      data['incorrect_information_explanation'] if info_correction_required?
+    end
+
+    def info_correction_required?
+      data['updates_needed'].include?('incorrect_information')
+    end
+
+    def resubmission_deadline
+      data['resubmission_deadline']
+    end
+
+    def sync_further_info_request
+      latest_further_info = data['further_information'].last
```

Comment:
For safety (there are no promises about what order this array might be in), might be worth doing `data['further_information'].max_by { _1['requested_at'] }`